src/io/parquet/read/file.rs

```diff
@@ -80,20 +80,21 @@ impl<R: Read + Seek> FileReader<R> {
             metadata: schema_metadata,
         };
 
-        let row_groups = RowGroupReader::new(
+        let mut row_groups = RowGroupReader::new(
             reader,
             schema,
             groups_filter,
             metadata.row_groups.clone(),
             chunk_size,
             limit,
         );
+        let current_row_group = row_groups.next().transpose()?;
```

Comment:
I think we should consider something different here - this causes `try_new` to be `O(N)` since it advances the iterator.